Unable to build docs on elixir master via `make docs`

I'm trying to build elixir docs locally, by git clone repo and then:

``` bash
make docs
==> elixir (compile)
mkdir -p ebin
rm -rf docs
cp -R -f lib/*/ebin/*.beam ./ebin
bin/elixir ../ex_doc/bin/ex_doc "Elixir" "0.9.1" -m Kernel -u "https://github.com/elixir-lang/elixir" --source-ref "v0.9.1"
** (UndefinedFunctionError) undefined function: ExDoc.CLI.run/1
    ExDoc.CLI.run(["Elixir","0.9.1","-m","Kernel","-u","https://github.com/elixir-lang/elixir","--source-ref","v0.9.1"])
    src/elixir_compiler.erl:65: :elixir_compiler."-eval_forms/4-fun-0-"/7
    src/elixir_compiler.erl:64: :elixir_compiler.eval_forms/4
    src/elixir_compiler.erl:32: :elixir_compiler.quoted/2
    /zfs/shared/repos/elixir/core/lib/elixir/lib/code.ex:234: Code.require_file/2
    /zfs/shared/repos/elixir/core/lib/elixir/lib/kernel/cli.ex:288: Kernel.CLI.process_command/2
    /zfs/shared/repos/elixir/core/lib/elixir/lib/enum.ex:604: Enum."-map/2-lc$^0/1-0-"/2
    /zfs/shared/repos/elixir/core/lib/elixir/lib/kernel/cli.ex:18: Kernel.CLI."-main/1-fun-2-"/1

make: *** [docs] Error 1
```

It seems you need to have ex_doc built and present in ../ex_doc -- which I couldn't find listed anywhere.

Once I corrected that, running `../ex_doc/bin/ex_doc`  manually still fails:

``` bash
bin/elixir ../ex_doc/bin/ex_doc                                             master✭
** (UndefinedFunctionError) undefined function: ExDoc.CLI.run/1
    ExDoc.CLI.run([])
    src/elixir_compiler.erl:65: :elixir_compiler."-eval_forms/4-fun-0-"/7
    src/elixir_compiler.erl:64: :elixir_compiler.eval_forms/4
    src/elixir_compiler.erl:32: :elixir_compiler.quoted/2
    /zfs/shared/repos/elixir/core/lib/elixir/lib/code.ex:234: Code.require_file/2
    /zfs/shared/repos/elixir/core/lib/elixir/lib/kernel/cli.ex:288: Kernel.CLI.process_command/2
    /zfs/shared/repos/elixir/core/lib/elixir/lib/enum.ex:604: Enum."-map/2-lc$^0/1-0-"/2
    /zfs/shared/repos/elixir/core/lib/elixir/lib/kernel/cli.ex:18: Kernel.CLI."-main/1-fun-2-"/1
```

with @true_droid's help on irc, it seems these are the correct steps:

```
# install elixir first
git clone git://github.com/elixir-lang/ex_doc.git &
git clone git://github.com/elixir-lang/elixir.git
(cd ex_doc && mix compile)
(cd elixir && make docs)
```

Perhaps this can go into either the README for elixir & ex_doc, or as a warning in the elixir Makefile if it can't find ex_doc in the correct path?

--- a/.gitignore
+++ b/.gitignore
@@ -14,3 +14,4 @@ erl_crash.dump
 .dialyzer_plt
 .dialyzer.base_plt
 .*.swp
+docs

--- a/Makefile
+++ b/Makefile
@@ -96,12 +96,15 @@ clean:
 
 SOURCE_REF = $(shell head="$$(git rev-parse HEAD)" tag="$$(git tag --points-at $$head | tail -1)" ; echo "$${tag:-$$head}\c")
 
-docs: compile
+docs: compile ../ex_doc/bin/ex_doc
 	mkdir -p ebin
 	rm -rf docs
 	cp -R -f lib/*/ebin/*.beam ./ebin
 	bin/elixir ../ex_doc/bin/ex_doc "Elixir" "$(VERSION)" -m Kernel -u "https://github.com/elixir-lang/elixir" --source-ref "$(call SOURCE_REF)"
-	rm -rf ebin
+
+../ex_doc/bin/ex_doc:
+	@ echo "ex_doc is not found in ../ex_doc as expected. See README for more information."
+	@ false
 
 release_zip: compile
 	rm -rf v$(VERSION).zip

--- a/README.md
+++ b/README.md
@@ -20,6 +20,15 @@ However, if tests fail, it is likely you have an outdated Erlang version (Elixir
 
 If you have the correct version and tests still fail, feel free to [open an issue][2].
 
+## Building documentation
+
+Building the documentation requires [ex_doc](https://github.com/elixir-lang/ex_doc) to be installed and built in the same containing folder as elixir.
+
+    # After cloning and compiling Elixir
+    $ git clone git://github.com/elixir-lang/ex_doc.git
+    $ cd ../ex_doc && ../elixir/bin/mix compile
+    $ cd ../elixir && make docs
+
 ## Contributing
 
 We appreciate any contribution to Elixir, so check out our [CONTRIBUTING.md](CONTRIBUTING.md) guide for more information. We usually keep a list of features and bugs [in the issue tracker][2].

